core/executor/src/precompiles/modexp.rs

```diff
@@ -72,6 +72,31 @@ impl PrecompileContract for ModExp {
     }
 }
 
+fn get_data(data: &[u8], mut start: usize, size: usize) -> Result<Integer, PrecompileFailure> {
+    let len = data.len();
+
+    if start > len {
+        start = len;
+    }
+
+    let mut end = start.wrapping_add(size);
```

Comment:
If this actually wraps, won't `data[start..end]` panic? How about using `saturating_add` instead.